compiler_opt/rl/corpus.py

```diff
@@ -125,4 +125,8 @@ def _load_and_parse_command(
 
   cmdline.extend(additional_flags)
 
+  # The options read from a .cmd file must be run with -cc1
+  if not cmd_override and cmdline[0] != '-cc1':
```

Comment:
are we attaching '-cc1' before your code change (i mean before you start the whole refactoring)? what made the difference that we don't do it before but do it now?

Reply:
before my changes, -cc1 was assumed to be in the .cmd files, if I use `Debian clang version 13.0.1-3+build2`, -cc1 is included in the .cmd

If I use Chrome's version of clang, which is near-tip-of-tree llvm, there's no -cc1 in the .cmd file

Reply:
seems like an llvm change probably, gonna go have a quick look

Reply:
I'd leave the options "as-is" - in some cases, the build system runs clang as a driver (despite running it on a single module); in others, the build system would run -cc1. Just adding -cc1 isn't necessarily correct, either.

Reply:
it's adding -cc1 iff we're reading off a .cmd file i.e. the .llvmcmd section, which must have been written by cc1-clang, which should've been called either by the clang-driver or directly, right?

anyway, turns out setting LLVM's DCMAKE_BUILD_TYPE to Debug eats the -cc1, somewhere, causing it to not end up in .llvmcmd

Reply:
settled on raising an exception instead of supporting this

Reply:
I just ran into this issue when testing out some stuff for the regalloc model. More specifically, the issue seems to occur when `LLVM_ENABLE_ASSERTIONS` is turned on, even when `CMAKE_BUILD_TYPE` is set to `Release`. I'm not sure what flags the chromium clang has enabled, but perhaps they have this turned on? Also seems like it's most likely a bug in clang/LLVM since toggling `LLVM_ENABLE_ASSERTIONS` shouldn't change behavior in this area. I'll try and look into this issue a little bit tonight while letting models train on the LLVM side.

Reply:
Just opened https://reviews.llvm.org/D130620. Should resolve the behavior with chromium if they're enabling assertions and also fix the debug/release behavioral differences.

Reply:
Yep, my Chromium toolchain builds have assertions on, thanks for looking into it & nice work on the fix!